src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1515,32 +1515,34 @@ public async Task<OrganizationLicense> GenerateLicenseAsync(Guid organizationId,
                     enoughSeatsAvailable = seatsAvailable >= usersToAdd.Count;
                 }
 
-                if (enoughSeatsAvailable)
+                if (!enoughSeatsAvailable) 
                 {
-                    foreach (var user in newUsers)
+                    throw new BadRequestException($"Organization does not have enough seats available. Need {usersToAdd.Count} but {seatsAvailable} available.");
+                }
+
+                foreach (var user in newUsers)
+                {
+                    if (!usersToAdd.Contains(user.ExternalId) || string.IsNullOrWhiteSpace(user.Email))
                     {
-                        if (!usersToAdd.Contains(user.ExternalId) || string.IsNullOrWhiteSpace(user.Email))
-                        {
-                            continue;
-                        }
+                        continue;
+                    }
 
-                        try
-                        {
-                            var invite = new OrganizationUserInvite
-                            {
-                                Emails = new List<string> { user.Email },
-                                Type = OrganizationUserType.User,
-                                AccessAll = false,
-                                Collections = new List<SelectionReadOnly>(),
-                            };
-                            var newUser = await InviteUserAsync(organizationId, importingUserId, user.Email, 
-                                    OrganizationUserType.User, false, user.ExternalId, new List<SelectionReadOnly>());
-                            existingExternalUsersIdDict.Add(newUser.ExternalId, newUser.Id);
-                        }
-                        catch (BadRequestException)
+                    try
+                    {
+                        var invite = new OrganizationUserInvite
                         {
-                            continue;
-                        }
+                            Emails = new List<string> { user.Email },
+                            Type = OrganizationUserType.User,
+                            AccessAll = false,
+                            Collections = new List<SelectionReadOnly>(),
+                        };
+                        var newUser = await InviteUserAsync(organizationId, importingUserId, user.Email, 
+                                OrganizationUserType.User, false, user.ExternalId, new List<SelectionReadOnly>());
+                        existingExternalUsersIdDict.Add(newUser.ExternalId, newUser.Id);
+                    }
+                    catch (BadRequestException)
+                    {
+                        continue;
```

Comment:
Wonder if this would need additional error handlings as well later on, will need to keep an eye on it as far as "what" could go wrong here that we'd just ignore.

Reply:
I looked into that from the same thought.

`BadRequestException` is thrown there if the user is already invited to the org. This is suggesting I should add a comment